share/management/commands/enforce_set_lists.py

```diff
@@ -0,0 +1,95 @@
+from share.management.commands import BaseShareCommand
+from share.models import SourceConfig, AbstractCreativeWork, ThroughTags
+from share.util import IDObfuscator
+
+
+class Command(BaseShareCommand):
+    """Enforce set whitelist and blacklist for the given source config(s)
+
+    Look at each source config's `transformer_kwargs` for `approved_sets` and `blocked_sets`.
+    Find and delete works in the database from that source that would not have been allowed,
+    according to those lists. If the work was ingested from multiple sources, remove the given
+    source from it instead of deleting.
+
+    By default enforce blacklists only, because enforcing whitelists is slooow.
+    Pass --whitelist to enforce whitelists too.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('source_configs', nargs='+', type=str, help='Labels of the source configs to look at')
+        parser.add_argument('--dry', action='store_true', help='Print changes that would be made, but make no changes')
+        parser.add_argument('--commit', action='store_true', help='If omitted, abort the transaction after making changes')
```

Comment:
What does aborting the transaction do? Not actually make the changes?

Reply:
Yeah, with neither `--dry` nor `--commit`, it'll actually call `administrative_change` and do stuff in the database, then roll it all back. It's useful to check that the changes won't give a database error without actually saving those changes.